### ISSUE TITLE
Flaky test: TestXSSWikiLastCommitInfo

### ISSUE THREAD

**Issue by silverwind** [go-gitea/gitea#29355]
Likely one of the recent changes triggered this.
````
--- FAIL: TestXSSWikiLastCommitInfo (0.46s)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user2/repo1.wiki.git/info/refs?service=git-upload-pack for 127.0.0.1:46238, 200 OK in 4.2ms @ repo/githttp.go:532(repo.GetInfoRefs)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed POST /user2/repo1.wiki.git/git-upload-pack for 127.0.0.1:46238, 200 OK in 4.4ms @ repo/githttp.go:492(repo.ServiceUploadPack)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed POST /user2/repo1.wiki.git/git-upload-pack for 127.0.0.1:46238, 200 OK in 12.1ms @ repo/githttp.go:492(repo.ServiceUploadPack)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user2/repo1.wiki.git/info/refs?service=git-receive-pack for 127.0.0.1:46250, 401 Unauthorized in 1.6ms @ repo/githttp.go:532(repo.GetInfoRefs)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user2/repo1.wiki.git/info/refs?service=git-receive-pack for 127.0.0.1:46250, 200 OK in 7.3ms @ repo/githttp.go:532(repo.GetInfoRefs)
    testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed POST /user2/repo1.wiki.git/git-receive-pack for 127.0.0.1:46250, 200 OK in 14.3ms @ repo/githttp.go:500(repo.ServiceReceivePack)
    --- FAIL: TestXSSWikiLastCommitInfo/Page_view (0.07s)
        testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user2/repo1/wiki/Home for test-mock:12345, 200 OK in 20.9ms @ repo/wiki.go:474(repo.Wiki)
        xss_test.go:109: 
              Error Trace:  /home/runner/work/gitea/gitea/tests/integration/xss_test.go:109
              Error:        Not equal: 
                            expected: "Gusted edited this page 0001-01-01 00:00:00 +00:00"
                            actual  : "Gusted <script class=\"evil\">alert('Oh no!');</script> edited this page 2024-01-31 00:00:00 +00:00"
                            
                            Diff:
                            --- Expected
                            +++ Actual
                            @@ -1 +1 @@
                            -Gusted edited this page 0001-01-01 00:00:00 +00:00
                            +Gusted <script class="evil">alert('Oh no!');</script> edited this page 2024-01-31 00:00:00 +00:00
              Test:         TestXSSWikiLastCommitInfo/Page_view
    --- FAIL: TestXSSWikiLastCommitInfo/Revision_page (0.07s)
        testlogger.go:60: 2024/02/23 22:34:04 ...eb/routing/logger.go:102:func1() [I] router: completed GET /user2/repo1/wiki/Home?action=_revision for test-mock:12345, 200 OK in 24.3ms @ repo/wiki.go:474(repo.Wiki)
        xss_test.go:121: 
              Error Trace:  /home/runner/work/gitea/gitea/tests/integration/xss_test.go:121
              Error:        Not equal: 
                            expected: "Gusted edited this page 0001-01-01 00:00:00 +00:00"
                            actual  : "Gusted <script class=\"evil\">alert('Oh no!');</script> edited this page 2024-01-31 00:00:00 +00:00"
                            
                            Diff:
                            --- Expected
                            +++ Actual
                            @@ -1 +1 @@
                            -Gusted edited this page 0001-01-01 00:00:00 +00:00
                            +Gusted <script class="evil">alert('Oh no!');</script> edited this page 2024-01-31 00:00:00 +00:00
              Test:         TestXSSWikiLastCommitInfo/Revision_page
FAIL
````

**Comment by wxiaoguang** [comment]
I think the test is not needed, could be deleted.

**Comment by wxiaoguang** [comment]
The test is not "flaky", but actually wrong.

->  Fix incorrect tests in 1.21 #29366
